### ISSUE TITLE
Patch development thread

### ISSUE THREAD

**Issue by ItaloKnox** [07th-mod/meakashi#1]
OKAY, not minding I did press enter instead of tab and submitted an empty thread, let me page our MVPs @enumag, @DoctorDiablo, @Grelo and @jwgrlrrajn.

Chapter is up, the scripts are already uploaded. I looked around and it seems to have some minor changes related to the previous chapter, still unsure about *what* really changed. You'll easily spot some unusual scripts in the dev folder. The main menu also changed, so I guess some more work will need to be done for the graphics.

For now it seems to be a really big chapter, around 60 thousand lines to work. Should take two weeks to finish if I do 5k a day. For now my plan is to go a little over that until my brain melts.

Anyway, discussion goes here. Have fun!

**Comment by Grelo** [comment]
I quickly checked how does graphic looks in CG folder and find out that names of bgs changed too. Now it seems they are named after their ps3 variants. Moreover i found lots of original bgs dont have their variants in ps3 graphic even main menu bg. This is gonna be fun xD

**Comment by DoctorDiablo** [comment]
@Grelo Do you want me to put together the basic sprites+backgrounds patch like I did last time, and you can make your changes on top of it?

**Comment by ItaloKnox** [comment]
In terms of compatibility we're fine. The syntax is still the same and the new scripts doesn't seem to interfere in anything. 

If you're worried about making your script changes while I'm working on the voice patch, we can easily merge both patches without any issues. In fact I prefer to merge it this way than having you work with a script with the voice patch already applied, a clean script minimizes the number of conflicts.

**Comment by enumag** [comment]
Oh my, that chapter is huge! I'm going to run the voice insertion script now. With a chapter of this size it will probably take over an hour. And there is the possibility that I run into some issue with the changes so it might take longer. I'll keep you posted.

**Comment by enumag** [comment]
@ItaloKnox All done. I need to redo the _meak_023.txt file though because the script inserted way too many wrong placeholders. Don't worry, I'll get it fixed before you get to it.

**Comment by Grelo** [comment]
@DoctorDiablo I think you dont need to. I think i need to create some bgs manually anyway. Ok hopefully it will not take forever... good luck.

**Comment by Grelo** [comment]
Preview of Meakashi scenario:
![scenario_c](https://cloud.githubusercontent.com/assets/22790067/25555009/7a3a1684-2cdc-11e7-934f-c9899d252edc.png)

Btw I will release the mod after its complete done with cgs, bgs etc.

**Comment by Grelo** [comment]
I decided ill be redoing everything even 4:3 images to 16:9 so this will be even better version than previous ....ill do the same with question arcs later after all is done.

**Comment by enumag** [comment]
@Grelo That sounds ambitous. And awesome. Consider me hyped. ;-)

**Comment by ItaloKnox** [comment]
The first batch is ready to go. Making placeholders for lines without voices was definitely the right choice, it was a lot more smoother than I expected. The amount of false positives wasn't that big and it happened exactly in places I expected, so it was pretty easy to fix.

**Comment by enumag** [comment]
@ItaloKnox Glad to hear that. Let me know if you run into anything you think the script should have handled better. Especially _meak_023.txt and _meak_024.txt seem to have more placeholders than expected (even with the reduction I made earlier). I also removed all placeholders from staffroom.txt - there were only placeholders but no matched voice files so I assume it is not meant to be voiced.

**Comment by Grelo** [comment]
Guys i need help. Do you know why the changes i made to script like adding cg are not showing in actual game? Even the changes which i deleted are and are not in actual script are showing..... 
Im stuck. I put it in the update folder btw.

**Comment by DoctorDiablo** [comment]
If you delete all the files in the CompiledUpdate folder, it will force the system to rebuild them.  If it doesn't generate one of the files, then there might be an error in one of them like a missing quote or something.

**Comment by Grelo** [comment]
I did that too but no help :(

**Comment by Grelo** [comment]
This is the modified version which i have in update folder

[_meak_001.txt](https://github.com/07th-mod/meakashi/files/966474/_meak_001.txt)

**Comment by jwgrlrrajn** [comment]
Yikes. They changed the file naming system for this one. Ugh. Now I have to manually compare images.

**Comment by ItaloKnox** [comment]
@Grelo try removing the file from the Update folder, clean both compiled folders and put it in the scripts folder (instad of Update). There's a very strange bug that happens rarely that the game just refuses to load things properly. By the way, if you are loading a save, that might also be your problem.

**Comment by DoctorDiablo** [comment]
@Grelo I think I found your problem.  It's on line 1008 in your updated script, it's missing the second parameter, "maskright" or whatever:

    DrawSceneWithMask( "black", 1, 0, 300 );

I copied your new script to the Update folder and it didn't load for me either.  So I looked in the output_log.txt in the HigurashiEp05_Data folder and it listed an error on that line.

`Failed to compile script _meak_001!`
`System.Exception: 1008: DrawSceneWithMask - Parameters were not of expected type!`

**Comment by Grelo** [comment]
@DoctorDiablo Wohoo, thanks that indeed was the issue. I can finally continue now.

**Comment by ItaloKnox** [comment]
If anyone here is feeling like doing idle work, can you guys check the "known issues" section and try to find the voices in the PS2 files? ``_meak_012.txt`` is probably the hardest one, but the others shouldn't take much time.

**Comment by enumag** [comment]
@ItaloKnox For the first mistake it seems that on PS2 there were these two voices, https://www.dropbox.com/s/cdnez8432vprf3m/150600178.ogg?dl=0 and https://www.dropbox.com/s/98kgrqd7b99iznv/150600179.ogg?dl=0. Later on PS3 the those two lines were combined to one https://www.dropbox.com/s/fmvf5ob2wdckek4/n_150600178.ogg?dl=0. At the moment the script has the PS2 lines but voices the PS3 variant. Which one do you prefer?

**Comment by ItaloKnox** [comment]
Thanks! Looks like they match perfectly. Since I was already with everything open, I committed the files and pushed them into the repo. Thanks again, I'll look for the remaining files after I finish the voice patch (probably tomorrow).

**Comment by enumag** [comment]
@ItaloKnox Actually I think your fix was not entirely correct so I have overwritten it with mine using force-push.

**Comment by ItaloKnox** [comment]
I guess you're right, though for some reason it looks okay on VS Code. 🤔

**Comment by enumag** [comment]
Wow it's such a big chapter but you're almost done already? I mean only a few days and the tips are left. You really work quickly. 👏

**Comment by ItaloKnox** [comment]
That's all thanks to the voice placeholders, couldn't go as fast if they weren't there. At first I was expecting 4 or 5 scripts a day (and that would be already pushing myself) but doing 8 a day was a lot more easier than before. I could even push it further and to it in two days, but since the graphics patch will take a while, I won't rush the release.

**Comment by Maelstrom1** [comment]
Speaking of the graphics patch, what's the timetable on that?

**Comment by enumag** [comment]
@Maelstrom1 We don't really have any at the moment, depends on @Grelo. However isn't quality more important than speed? ;-)

**Comment by Grelo** [comment]
Guys im giving it my all. Its easily the hardest chapter for now. Maybe tommorow ill have already half done.

**Comment by ItaloKnox** [comment]
Don't worry about the speed, if we can get it finished in two weeks I think we're good. The voice patch always come out early so if anyone is eager to play they can get that. The more time we take to finish it = less bugs we have to squish.

**Comment by Grelo** [comment]
True :) . Anyway do you guys know why meakashi dont have textbox? Its displaying on some filtered screen and it doesnt look good.

**Comment by ItaloKnox** [comment]
You mean the vanilla game? I always thought every 07th Expansion game had no textboxes, it was always the plain transparent black filter.

**Comment by Grelo** [comment]
So will you make it on release, or can i make it someway? Dont need it anyway now.

**Comment by enumag** [comment]
@Grelo I think you just need to replace the windo_filter.png in the CG directory with one from our older patches. Plus the init.txt script in the Update directory. That should do the trick I think.

**Comment by Grelo** [comment]
@enumag thanks it works, so easy, didnt thought it would be in cg folder.

**Comment by Grelo** [comment]
How does the new unlock scenarios look? I will make similar ones for question arcs in the future.

![omake_jamp](https://cloud.githubusercontent.com/assets/22790067/25594361/763dfc86-2ec0-11e7-9b1b-127b9c68a7b6.png)
![omake_staff](https://cloud.githubusercontent.com/assets/22790067/25594362/763f6ee0-2ec0-11e7-9c49-485ac2b7c84e.png)
![Uploading omake_tip.png…]()

**Comment by enumag** [comment]
@Grelo I only have one word to describe them: gorgeous. 🥇

**Comment by Grelo** [comment]
Btw i made HD versions of notes in original folder too. They are displaying on whole screen now. Dont know if its good or bad.

**Comment by enumag** [comment]
I prefer 16:10 :trollface:.

It is better to have it consistent I think? It doesn't matter much for the notes though. I mean it's just blackscreen anyway.

**Comment by Grelo** [comment]
Example:
![58b](https://cloud.githubusercontent.com/assets/22790067/25594806/8475649a-2ec2-11e7-9780-674499864313.png)

**Comment by ItaloKnox** [comment]
For the notes it doesn't matter that much, but I still think it's better for people with high resolution displays. If we're going to have the best version, then I think it's a good effort.

By the way, the scenario screens look beautiful. Great work!

**Comment by DoctorDiablo** [comment]
Wow, _meak_012.txt was missing a lot of voice files.  I found them all (I think) in the PS2 sound files, so I should be able to add them in pretty easily.

And nice work on the graphics, @Grelo!

**Comment by ItaloKnox** [comment]
Yeah, the PS3 release is a shitfest compared to the PS2 version. So much stuff changed or removed, sometimes I prefer to leave voice files out instead of censoring the text. For now I only changed (and translated) two sections, and I hope it stays this way. Anyway, if you can get the voice files in place for me, I would really appreciate that.

**Comment by DoctorDiablo** [comment]
I fixed all the voice issues except the one that mentioned changing the translation, which sounded a little scary.

**Comment by ItaloKnox** [comment]
Thanks, @DoctorDiablo, I'll take a second look at that to see what is the best route. There's now a few (a lot?) more, so if you can do these as well I would be grateful. If you can't, no problem, I'll come back to them after tomorrow.

Also @enumag, can you look at the newest commit? I'm not sure what the hell happened, but for some reason the voice inserter thought it was good to put a bunch of S02 voices in the last three scripts. It was kinda hellish and took a lot of time. 

Maybe for a future version can you include a blacklisting feature? Use the first two scripts as a sample, get the most common folder (in this case it was S06 for the whole game -- but dunno about the tips) and use it until the end.

**Comment by enumag** [comment]
@ItaloKnox Sorry about that. I'll look into it tomorrow and hopefully find some solution. Blacklisting based on the first few scripts doesn't feel right - what if an error like this occured in those.

Also I wanted to ask ane more thing. How often does the script insert a voice file somewhere outside of a dialogue? Should I disable insersion completely when outside of quotes?

**Comment by Grelo** [comment]
@ItaloKnox Dont change the text pls :( . Btw graphic patch progressed a lot so.....soon :P

**Comment by ItaloKnox** [comment]
@enumag It's not actually that common to see voices outside of a dialog. I actually prefer the way it is right now, the only issue I had was with the S02 fiesta that slowed me down a lot. Don't feel bad about it, what we have right now is the best, it flows better than ever.

@Grelo tbh I'm very against changing the translation, so I only do that in very minor issues. In this chapter there are a lot of graphic scenes, so no way I'm changing them or removing them because we can't have them voiced. Some minor stuff like copyrighted references or the last one where I changed from "are you drunk" to "did you put an all nighter" (so the voice line would match, of course) are ok with me. As long as it doesn't change the meaning, remove content or make it less fun I think it's fine. But if anyone disagrees, we can always discuss it, since the patch is a collaborative effort and everyone has a say on it.

**Comment by DoctorDiablo** [comment]
@ItaloKnox I fixed some more of the missing voices.  There were two voices you said were censored and didn't match the text, but the PS2 audio files sounded the same as the PS3 version (at least as far as I could tell?) so you can double-check those.

**Comment by Grelo** [comment]
@ItaloKnox If the meaning is preserved its ok or the way you described it. I meant removing big chunks of text due to missing voices.

**Comment by enumag** [comment]
@ItaloKnox I see _meak_024.txt is the only file left. I removed all s02 lines from my database and runned the inserter again for this file only. It looks much better now.

Unfortunately this works for this case only. I was unable to fix it reliably without manually releting all s02 lines. We will need to be careful about similar cases in the future arcs.

**Comment by Grelo** [comment]
How is going progress guys? I hope you will finish it soon. I fully finished graphic mod. Will upload it today.

**Comment by ItaloKnox** [comment]
@enumag Thanks a lot, I'm going to start working on it right now. If anything odd happens I'll let you know.

@Grelo The last batch of the voice patch should come out today, so after that I'll have to squish the remaining bugs and then release the voice patch. After that we'll merge it with the graphics patch and release the real thing. Good to know that you have already finished it.

**Comment by Grelo** [comment]
Ok guys, here is the mod. I worked really hard on it, ufff. So a few things. This is how the mod should look.
Its 100% standalone that means if you plan to merge it with original CG folder it will leave a lot of unnecessary pics in the folder like the installer we are using now. Next You already know its FHD but if someone doesnt like it i apologize. The names of the files are mostly changed and i havent created sunset,night, etc. folders, but im confident enough to say there are no "real" issues with the mod and it should be working at least on 98%, i tried to QC it. So im now waiting for voices to be merged in it and enjoy it anew. Enjoy!

https://mega.nz/#!tY8DmZ6A!6HJwiOcmQtFo9JwPA1hyqAC81GSWjIk38r89B9TQ1cQ

**Comment by jwgrlrrajn** [comment]
FHD?

**Comment by ItaloKnox** [comment]
My turn now, the final batch is already finished and uploaded. Now the last thing to do is to find some missing voices and release the voice patch, then merge and release the voices and graphics patch. Great work, @Grelo!

@jwgrlrrajn Full HD!

**Comment by jwgrlrrajn** [comment]
Ah. Got it.

**Comment by ItaloKnox** [comment]
The bugs are now fixed, some stuff was also censored in the PS2 voices but it's really minor. Just one small part.

Can I leave two missions for you, @enumag? Update the compiler so we can get the voices for Meakashi and merge the graphics patch.

**Comment by enumag** [comment]
Sure.

**Comment by jwgrlrrajn** [comment]
Oh dear. Seems making a compatible MG patch is going to be harder than I thought. Before, I was just able to copy the folder over to the existing CGAlt folder, avoid overwriting any existing files, then just searching for any PS3 sprites that were needing to be replaced with altered MG sprites. Here, I don't know what's duplicated and what isn't, since only 14 files had the same name.

Also, Sonozaki Akane has three sprites in the PS3 version, while MG only has one, and there aren't any other games to get the extra expressions from yet. So, while I have this...
![aka_def](https://cloud.githubusercontent.com/assets/13456824/25676437/9eff39ee-3007-11e7-8001-26307ad2c0cc.png)

I don't have a replacement for this...
![aka_sakebi_0](https://cloud.githubusercontent.com/assets/13456824/25676451/a94814f2-3007-11e7-8ea7-341ae6aaf67d.png)

**Comment by ItaloKnox** [comment]
Oh, by the way @enumag, when you commit the merged patch just commit it inside a new Update folder. I'll leave the dev folder for the voices-only patch.

@jwgrlrrajn in this case I can only think of two ways:
1. just use that one sprite (doesn't seem to be a major issue)
2. use the PS3 sprites instead...

I would take the first option without thinking twice, though. Doesn't looks like a big issue.
Wish we had someone with mad drawing skills to remake the sprites, but guess that's asking too much.

**Comment by Grelo** [comment]
@jwgrlrrajn Sorry for that weird file names, but MGs naming often didnt have variant in PS3 so i was changing them on the go to be easier for me :(

**Comment by enumag** [comment]
@jwgrlrrajn In my opinion the MG patch should use MG sprites as much as possible even if they lack the expressions.

@ItaloKnox Understood!

**Comment by enumag** [comment]
Here is the list of voices that are used in the scripts but probably should not be. We need to fix these before compiling the voice patch. Some of theme, especially S02 need to be replaced with their S06 equivalents so that's probably for you @ItaloKnox.

```
SE/S02/00/132600004
SE/S02/00/132600006
SE/S02/03/130300185
SE/S02/03/130300728
SE/S02/03/130300729
SE/S03/01/140100570
SE/S05/13/101300625
SE/S13/00/550300011
SE/S19/10/501000058
```

**Comment by jwgrlrrajn** [comment]
So, just have three copies of the file with different names then? Makes enough sense, given Akane probably only had one expression in the original anyway (She'd be easier to edit if she was facing the screen directly and didn't have hair covering her eyebrows, but it is what it is).

**Comment by enumag** [comment]
@jwgrlrrajn Yeah, three copies of the same file. It's unfortunate but we don't have much of a choice here. Let's hope she will get more expressions in the future.

**Comment by ItaloKnox** [comment]
@enumag Oops, I don't usually check the folder but sometimes the S02 voices have the same filename as the S06 voices, guess that's why I missed them. I searched the whole thing for S02 voices and fixed those I found, thanks for reporting that.

**Comment by ItaloKnox** [comment]
Since we're already doing a lot, I went around and cleaned a bit the other repositories. Lots of unused stuff and the readmes are talking about stuff we don't need anymore. Something else caught my attention: is Watanagashi that problematic? The known issues section is massive.

After we release the Meakashi patch I'll probably come back to Watanagashi and try to fix most of the stuff.

Oh, and something else, what you guys think about creating a logo for our group and use it in our releases? Like the 07th Expansion and MangaGamer splash screens at the start. I also think it's a good idea because the user can instantly know if the patch has been successfully installed. I want to hear your opinions about this, maybe we can try to put something together and discuss which one is the best.

**Comment by DoctorDiablo** [comment]
I like the logo idea.  It makes the product look more official.

@Grelo I did some testing with your patch, and it looks really good.  So far I've only found one really minor issue, where Irie's sprite is being [drawn on top](http://imgur.com/a/NXTqV) of Satoshi's zoomed sprite.

One suggestion though: You created a bunch of grayscale/sepia/negative image files, but a lot of that functionality can be done programmatically by modifying the script to use the DrawFilm() and Negative() commands.  Might be easier to do it that way in the future.

**Comment by jwgrlrrajn** [comment]
If you give me a couple of days, I might be able to come up with a logo candidate after finals are over. What would we want the logo to say though?

**Comment by ItaloKnox** [comment]
Just the name of the group, 07th Modding. It doesn't have to be something really flashy or fancy, just something that looks nice.

I'll try some stuff as well, let's see what we can make.

**Comment by ItaloKnox** [comment]
I said it doesn't have to be flashy or fancy but you know, fuck it. We should make whatever looks cool.

I had this random idea while I was looking the sprites, what do you guys think?
![07thmod](https://cloud.githubusercontent.com/assets/4702556/25687757/822c46a2-304f-11e7-8501-f44181c0064b.png)

Here how it looks when you open the game: https://gfycat.com/GlamorousFlusteredHind
I thought about making each sprite appear at a time but it seems to take too much time. Cool to watch once but not cool every time you open the game.

**Comment by Grelo** [comment]
@DoctorDiablo If you mean the corner of Satoshis shirt is behind Iries sprite, i couldnt fix that, dont know how i tried everything but its very minor, normal person will not even spot that. But if you know how, fix it please thanks.

**Comment by Grelo** [comment]
About logo do you want character image in background or not?

**Comment by ItaloKnox** [comment]
No preference, just make whatever you likes the most. As long as it's nice.

**Comment by ItaloKnox** [comment]
Hey @enumag, I tried running the compiler but it is bugging like the last time, it runs the whole thing and deletes all the voices. This time I tried it on Windows, I got a Windows Server 2008 running and it seems to be fine with the previous chapters. Any ideas?

**Comment by Grelo** [comment]
I made this. You can throw it away if you want.

![7th modding logo](https://cloud.githubusercontent.com/assets/22790067/25705487/c1e37994-30dd-11e7-9c2c-bc40aa2a6670.png)

**Comment by Grelo** [comment]
My creation. Feel free to not use it.

![7th modding logo](https://cloud.githubusercontent.com/assets/22790067/25707606/5bdd8b24-30e4-11e7-9fee-d96a05cfeabc.png)

Im already impatient to play Meakashi with voices + graphics ✌️

**Comment by enumag** [comment]
Oh my. I'm gone for less than a day and there are so many comments. It's really great how things are moving here. :-)

Watanagashi is the only chapter I checked thoroughly so that's probably where those issues came from. I'm sure I'll find some in the other chapters as well when I finally get to read them.

Logo idea is good I would not want to watch the three logos as in your gif every time I start the game. What about combining all three to a single screen?

Compiler won't work for Meakashi until the scripts are in the Update directory - which means I need to merge the voices and the graphics patches which was blocked by the wrong voices I posted above. There are some conflicts I need to resolve but I should upload the result today.

**Comment by enumag** [comment]
The patches are merged. I also fixed the case sensitivity issues so the compiler will now work on linux as well. It will report missing graphics because I'm not done with them yet but since you only need it to delete the unused voices you can just ignore that.

Also run it like this (with --force) to redownload the resources:

```
php console.php higurashi:make meakashi --force
```

**Comment by ItaloKnox** [comment]
I feel like three logos aren't that much of a problem since they don't take long to appear and if you really want to play asap, you can click or hold ctrl to skip them.

By the way, thanks for merging and updating the compiler. Already uploaded the voices and now all that's left is to click the release button for the voices patch. I'll release the voices and graphics patch right after I run the installer and test it.

**Comment by ItaloKnox** [comment]
I guess that's it. The installer is working, the game seems to run fine. We're all done for now.

Pop open the champagne, folks!

(unless I goofed somewhere and forgot about it)

**Comment by enumag** [comment]
@ItaloKnox Wow. It's not even a week since we started and we already have everything except the MG patch done and ready. Plus considering all the improvements in our development process it should be far less buggy then our previous patches were for their first release. Great work @ItaloKnox and @Grelo!

**Comment by DoctorDiablo** [comment]
Awesome! I'll give it a proper play through and try to find any leftover bugs.

**Comment by DoctorDiablo** [comment]
Ran into an error during the install.  [Looks like it couldn't download one of the archives](http://imgur.com/a/KuDL8) so I ended up with an empty Update folder.

**Comment by ItaloKnox** [comment]
Fuck, that's the wrong version. Give me a second.

**Comment by ItaloKnox** [comment]
Done, it should work now. I forgot to change v1.2 to v1.0.

**Comment by DoctorDiablo** [comment]
Thanks, it's working now.

**Comment by ItaloKnox** [comment]
There was a update for Meakashi that updated some of the scripts. Most likely typo fixes, I guess.

Anyone feeling like running a diff?

**Comment by enumag** [comment]
@ItaloKnox All merged.

**Comment by ItaloKnox** [comment]
You're the best. I'll release a new version tomorrow.

**Comment by DoctorDiablo** [comment]
My last commit just crapped on all the files.  I must have changed the encoding or something.  Is there an easy way to roll back a commit?

**Comment by enumag** [comment]
@DoctorDiablo Reverted and repushed correctly https://github.com/07th-mod/meakashi/commit/5b3d468c7bf06461ab6984e87d17ba71d61eba36. It was probably a CRLF issue.

**Comment by enumag** [comment]
@DoctorDiablo If you're using windows you should use `git config --global core.autocrlf true` to make git commit all files with linux line endings. That should fix the issue.

**Comment by DoctorDiablo** [comment]
All right, I finished reviewing Meakashi.  I fixed a few minor spelling issues, and added a couple missing voices from the PS2 version.  Otherwise, everything was perfect.  You've got my stamp of approval!

**Comment by ItaloKnox** [comment]
Nice, I took this opportunity to push a new version of the patch with your fixes. Should be the last one for a while until we can accumulate solved issues or anything critical comes up. For now I guess I should focus on having the new graphics patches uploaded.

**Comment by Maelstrom1** [comment]
Is there a mangagamer sprite patch in the works?

**Comment by enumag** [comment]
@Maelstrom1 https://github.com/07th-mod/meakashi/issues/1#issuecomment-299003186

**Comment by Maelstrom1** [comment]
Oh. Thanks

**Comment by jwgrlrrajn** [comment]
Working on getting the Meakashi MG patch out here. Not sure what I need to do though. As I understand it, the PS3 sprites go in CG and the MG sprites go in CGalt, but since the file names are different, do I need to change the MG file names to patch the PS3 file names?

**Comment by enumag** [comment]
@jwgrlrrajn I'm afraid that's the case.

**Comment by ItaloKnox** [comment]
That also reminds me to ask @Grelo about all the unused files in (mostly old sprites and BGs) the CG folder. Is it okay to purge the whole folder before extracting the patch in?

**Comment by Grelo** [comment]
@ItaloKnox In Meakashis case yes.

**Comment by enumag** [comment]
@Grelo Actually we're pretty much done with the question arcs. All that's left is uploading the updated MG patches and changing the installers.

**Comment by ItaloKnox** [comment]
The installers are already updated, all that's left is to push them and get @jwgrlrrajn to publish the patches on our resources repo (he can do that himself or I can do with my server).

**Comment by Grelo** [comment]
Well done, then. So whats left? Fonts and everything around it? Maybe the opening?

**Comment by enumag** [comment]
Pretty much. Then maybe updating resources outside of CG/CGAlt if we can figure out how. And the issues with console arcs. https://github.com/07th-mod/higurashi-console-arcs/issues

**Comment by jwgrlrrajn** [comment]
Took me longer than I thought, but here's my 07th Modding logo screen for Higurashi if you're interested (Now, if only Higurashi had some kind of iconic image I could use for an Umineko version :stuck_out_tongue: ).
![07th modding](https://cloud.githubusercontent.com/assets/13456824/26026291/46eab3d2-37be-11e7-8e87-0e1eddf8b543.png)

**Comment by jwgrlrrajn** [comment]
Here's a proof of concept for Onikakushi...
![07th modding oni](https://cloud.githubusercontent.com/assets/13456824/26030236/7b1653ce-3812-11e7-8aeb-202cc8208c61.png)

**Comment by ItaloKnox** [comment]
I just updated the installer to enable the updating feature. It should now install a new .exe that allows you to pull the updates individually. I'm working on something simple to keep tracking of what version the user installed and the remote version, just to avoid updating when there's no update.

Also a minor change but the installer should now use a PS3 icon. I'll leave it alone for Meakashi for now and update the others retroactively after we get enough testing data. I'll also try to squish some current bugs to avoid updating all installers at once.

**Comment by enumag** [comment]
@ItaloKnox Do we want to do something with the images @jwgrlrrajn posted above?

**Comment by ItaloKnox** [comment]
Let's put it on hold for now, I want to focus on the dll improvements.
